src/algebra/algebra/spectrum.lean

```diff
@@ -268,10 +256,32 @@ variables [comm_ring R] [ring A] [algebra R A]
 local notation `σ` := spectrum R
 local notation `↑ₐ` := algebra_map R A
 
-theorem left_add_coset_eq (a : A) (r : R) :
-  left_add_coset r (σ a) = σ (↑ₐr + a) :=
-by { ext, rw [mem_left_add_coset_iff, neg_add_eq_sub, add_mem_iff],
-     nth_rewrite 1 ←sub_add_cancel x r, }
+-- it would be nice to state this for `subalgebra_class`, but we don't have such a thing yet
+lemma subset_subalgebra {S : subalgebra R A} (a : S) : spectrum R (a : A) ⊆ spectrum R a :=
+compl_subset_compl.2 (λ _, is_unit.map S.val)
+
+-- this is why it would be nice if `subset_subalgebra` was registered for `subalgebra_class`.
+lemma subset_star_subalgebra [star_ring R] [star_ring A] [star_module R A] {S : star_subalgebra R A}
+  (a : S) : spectrum R (a : A) ⊆ spectrum R a :=
+compl_subset_compl.2 (λ _, is_unit.map S.subtype)
+
+lemma singleton_add_eq (a : A) (r : R) : {r} + (σ a) = σ (↑ₐr + a) :=
+ext $ λ x,
+  by rw [singleton_add, image_add_left, mem_preimage, add_comm, add_mem_iff, map_neg, neg_neg]
+
+lemma add_singleton_eq (a : A) (r : R) : (σ a) + {r} = σ (a + ↑ₐr) :=
+add_comm {r} (σ a) ▸ add_comm (algebra_map R A r) a ▸ singleton_add_eq a r
+
+lemma neg_eq (a : A) : -(σ a) = σ (-a) :=
+set.ext $ λ x, by simp only [mem_neg, mem_iff, map_neg, ←neg_add', is_unit.neg_iff, sub_neg_eq_add]
+
+lemma singleton_sub_eq (a : A) (r : R) :
+  {r} - (σ a) = σ (↑ₐr - a) :=
+by rw [sub_eq_add_neg, neg_eq, singleton_add_eq, sub_eq_add_neg]
```

Comment:
Could you also make a `r -ᵥ (σ a)` version?

Reply:
It seems we have `set.has_vadd_set`, but there is no analogous `set.has_vsub_set`, so I didn't add this version.